### ISSUE TITLE
plotTracks throwing an error, cannot find 'x' on as.data.frame

### ISSUE THREAD

**Issue by mwodring** [ivanek/Gviz#82]
I haven't got a minimum example since I can't even tell you what's wrong. To summarise, I'm plotting a few tracks; one with coverage, one with ORFs, and then a details track that names the domains found in those ORFs. So far the one contig that keeps throwing this error isn't missing any of those things. Other contigs are and I just filter those out. But this one contig is quite stubborn! I'm not expecting you guys to troubleshoot my evil contig for me, but I'm wondering where data frames come in?

I'd mainly like to know if the devs have any way 'around' the empties? Were exactly does it become a data frame and what can I do to tell it to fill the empties with a string of some kind? For now I've had to wrap it in a tryCatch telling people to check out the contig that threw the error, but it would be nice to plot it with NAs, since that's what the script does if it finds empty ORFs i.e. no matches. It's definitely plotTracks causing the trouble. If you like I have a saved image of the individual tracks just before the error showed up but I don't think it's an 'error'. Just me not quite understanding.

**Comment by mwodring** [comment]
![Example_output](https://github.com/ivanek/Gviz/assets/89640991/f89ad30f-b16b-4141-9714-a3c86e1c5f90)

When it works it should look like so. Like I said, I don't think it's an error with the software I just don't think there's a way to submit a question here that isn't an 'issue'.

**Comment by ivanek** [comment]
Hi @mwodring,
Would be possible for you to share the script and example data? 
Thanks, Robert

**Comment by mwodring** [comment]
> Hi @mwodring, Would be possible for you to share the script and example data? Thanks, Robert

Unfortunately, I'm not sure I can. The files involved are quite large and I'm not sure which parts of which are causing the error. I can provide the script though; would you prefer a ZIP here or a link to a GIST?

**Comment by ivanek** [comment]
You can put the script in gist or attach it here, whatever you prefer. 
Can't you subset only specific region(s) from the input files? Without it it would be hard to debug it.

**Comment by mwodring** [comment]
Thanks for your suggestion. I had some difficulty but I subsampled 5000 reads and this is causing a different error but again, only on some files.

The problem was there's a datatrack trying to plot using an empty dataframe since there's an empty dataframe on one iteration of the loop. That's not an issue with this software I think. I grepped the sample name in the file it's looking for the sample in and it's in there so now I'm just confused.

I get: rpy2.rinterface_lib.embedded.RRuntimeError: Error in ans[npos] <- rep(no, length.out = len)[npos] : incompatible types (from S4 to logical) in subassignment type fix

Is the error when a datatrack is given something empty. So I think this isn't an issue with this unless you want to somehow clarify the error message from being passed something empty? But that's up to you of course. Thanks for your time though and I'm sorry the error was further upstream than I thought it was; at the time the error was on plotTracks and I couldn't debug further.
